docs/3.0rc/develop/write-tasks/index.mdx

```diff
@@ -483,148 +483,223 @@ def my_flow():
+
+All of these are hashed to compute the task's _cache key_.
+
+### Cache keys
+
+To determine whether a task run should retrieve a cached state, Prefect uses the concept of a "cache key". 
+A cache key is a computed string value that determines where the task's return value will be persisted within
+its configured result storage.
+When a task run begins, Prefect first computes its cache key and uses this to lookup a record in the task's result
+storage. 
+If an unexpired record is found, this result is returned and the task does not run but instead enters a 
+`Cached` state with the corresponding result value.
+
+Cache keys can be shared by the same task across different flows, and even among different tasks, 
+so long as they all share a common result storage location.
+
+By default Prefect stores results locally in `~/.prefect/storage/`. 
+The filenames in this directory will correspond exactly to computed cache keys from your task runs.
 
 <Warning>
-**Task results and caching** 
+**Relationship with result persistence** 
 
-Task results are cached in memory during a flow run and persisted to your home directory by default. 
-Prefect Cloud only stores the cache key, not the data itself.
+Task caching and result persistence are intimately related. Because task caching relies on loading a 
+known result, task caching will only work when your task can persist its output 
+to a fixed and known location.
+
+Therefore any configuration which explicitly avoids result persistence will result in your task never
+using a cache, for example setting `persist_result=False`.
 </Warning>
 
-Caching refers to the ability of a task run to reflect a finished state without actually running the 
-code that defines the task. This allows you to efficiently reuse results of tasks that may be expensive 
-to run with every flow run, or reuse cached results if the inputs to a task have not changed.
+### Cache policies
 
-### Cache keys
+Cache key computation can be configured through the use of _cache policies_. 
+A cache policy is a recipe for computing cache keys for a given task.
 
-To determine whether a task run should retrieve a cached state, we use "cache keys". A cache key is a 
-string value that indicates if one run should be considered identical to another. When a task run with a 
-cache key finishes, we attach that cache key to the state. When each task run starts, Prefect checks for 
-states with a matching cache key. If a state with an identical key is found, Prefect uses the cached 
-state instead of running the task again.
+Prefect comes prepackaged with a few common cache policies:
+- `DEFAULT`: this cache policy uses the task's inputs, it's code definition, as well as the prevailing flow run ID
+to compute the task's cache key.
+- `INPUTS`: this cache policy uses _only_ the task's inputs to compute the cache key.
+- `TASKDEF`: this cache policy uses _only_ the task's code definition to compute the cache key.
```

Comment:
FWIW, all the other values here are self documenting but this one really isn't clear unless you read this doc. 

Suggestions:

`TASK_CODE`
`TASK_DEF` (maybe)
`TASK_SOURCE`

Reply:
Love it; `TASK_SOURCE` sounds best to me but it's a bit formal. Will need to think about which to use but I'll change it either way

Reply:
source actually hints at how I get this (`getsource()`) so I think it's the best

Reply:
Made this change in my code companion PR to this: https://github.com/PrefectHQ/prefect/pull/14164/commits/e8c996d0d5a391bb8f11118c5cd59a07665c494d